### DISCUSSION TITLE
Multiple table columns of a database column

### DISCUSSION THREAD

**Discussion by Wulfheart** [filamentphp/filament#5477] [Help] [answered]
Currently, if I declare multiple columns which are based on the same model attribute then only the last one gets printed in the table:
````php
TextInputColumn::make('sc_count')->label(__('core/player.attributes.sc_count'))
  ->type('number')
  ->rules(['integer', 'min:0']),
Tables\Columns\TextColumn::make('sc_count')->label(__('core/player.attributes.sc_count')),
Tables\Columns\TextColumn::make('sc_count')->label(__('core/player.attributes.sc_count')),
````

However, if I declare them mutually exclusive, then no column gets shown:
````php
TextInputColumn::make('sc_count')->label(__('core/player.attributes.sc_count'))
  ->type('number')
  ->rules(['integer', 'min:0'])
  ->hidden($gameFinishedCallback),
Tables\Columns\TextColumn::make('sc_count')->label(__('core/player.attributes.sc_count'))->visible($gameFinishedCallback),
````

Is there any way I can make this work that does not involve hooking giving the columns different names and do some lifecycle hooks?

Thank you in advance.

**Comment by danharrin** [comment, marked as answer]
Use columns with different names, and then use `getStateUsing(fn ($record) => $record->sc_count)` etc

**Reply by Wulfheart** [reply]
Nice. Thank you!
